### ISSUE TITLE
-r support for Mach-O

### ISSUE THREAD

**Issue by XrXr** [bluewhalesystems/sold#12]
I recently made Ruby's YJIT use ld64's `-r` to cut down the list of unintentionally exported symbols:

https://github.com/ruby/ruby/blob/da9ee7bcf361887a28a1dd6769a4f47261dea7aa/yjit/yjit.mk#L58-L74

I tried using sold with this setup, but I got `mold: fatal: unknown command line option: -r` as of df8b71d38a575376517411a6cb56962a8aac2433.

It would be nice if sold supports this option. No one linker seems to be perfect for this setup it seems. ld64 seems to have a [bug](https://developer.apple.com/forums/thread/723426) that prevents hiding all the symbols I want to hide, and LLD doesn't support `-r` at the moment.

**Comment by rui314** [comment]
Yup, we need to add a support for `-r`. That's on my to-do list.
